File: app/schemas/organization.py

from pydantic import BaseModel, ConfigDict
from typing import Optional, List


# Forward declaration for recursive schema
class Organization(BaseModel):
    id: int
    name: str
    level: int
    department_grade: Optional[str] = None
    parent_id: Optional[int] = None
    children: List["Organization"] = []

    model_config = ConfigDict(from_attributes=True)

# Update forward reference
Organization.model_rebuild()

class OrganizationCreate(BaseModel):
    name: str

    level: int
    parent_id: Optional[int] = None
    department_grade: Optional[str] = None


class OrganizationUpdate(BaseModel):
    name: Optional[str] = None
    level: Optional[int] = None
    parent_id: Optional[int] = None
    department_grade: Optional[str] = None

File: frontend/src/schemas/organization.ts
import { DepartmentGrade } from "./evaluation";

export interface Organization {
  id: number;
  name: string;
  level: number; // Renamed from organization_level
  parent_id: number | null;
  department_grade?: DepartmentGrade;
}

export interface OrganizationCreate {
    name: string;
    level: number; // Renamed from organization_level
    parent_id?: number | null;
}

export type OrganizationUpdate = Partial<OrganizationCreate>;